src/backend/distributed/operations/shard_transfer.c

```diff
@@ -1170,6 +1173,11 @@ CopyShardTables(List *shardIntervalList, char *sourceNodeName, int32 sourceNodeP
 		CopyShardTablesViaBlockWrites(shardIntervalList, sourceNodeName, sourceNodePort,
 									  targetNodeName, targetNodePort);
 	}
+
+	/*
+	 * Drop temporary objects that were marked as CLEANUP_ALWAYS.
+	 */
+	FinalizeOperationNeedingCleanupOnSuccess("citus_[move/copy]_shard_placement");
```

Comment:
since this string shows up in error messages, it would be slightly nicer if we passed down the operation name like we do in split variants.